### DISCUSSION TITLE
Connecting the bord to RJ12 to P1

### DISCUSSION THREAD

**Discussion by devdems** [UtilitechAS/amsreader-firmware#791] [Q&A] [answered]
Hi,

i have this ESP32 board https://cdn.shopify.com/s/files/1/1509/1638/files/ESP32_DevKit_C_V4_Pinout.pdf?v=1615364529

As you see my TX port is GPIO1 and RX port is GPIO3.

I'm using ams2mqtt-esp32-2.3.5 and the meter i wan't to connect to is Iskraemeco AM550.

1. Do i need to configure HAN RX and HAN TX if i will be using P1 connection?

     - If I do need to configure this port i have a problem with selecting GPIO1 for TX as I don't have this port available in dropdown.

2. Does someone have any diagram on how to connect RJ12 pins to the board?

Tnx!

**Comment by ArnieO** [comment]
1. The DATA output of a meter with P1 interface is an Open Collector output. You need to install a pullup resistor on it to see anything. 3k is a suitable value to pull it to 3,3V.
2. You can use any available GPIO except Rx/Tx, that are used for outputting diagnostics. 
3. You will probably need to power your board from USB during operation. If you rely on the 5V output from the meter you risk running into issues. (I can elaborate if you want to).
4. Which country are you in? The Iskraemeco AM550 seems to be in use in several countries with varying payload formats. We do not have full overview, and this is unfortunately not standardized. It could be that we need to do a modification on the firmware to accommodate the format from your meter. Search under "Issues" here to find examples.

**Comment by devdems** [comment]
@ArnieO thank you!

So basically like this just with a bigger resistor of 3k ohm?

![image](https://github.com/UtilitechAS/amsreader-firmware/assets/8215353/721e3123-5dae-451f-b0ef-9c6d0a544d1f)

Powering the device with external USB supply in my case is not possible so I need to rely on 5v from the meter. What can I expect?

I'm from Slovenia and from the commits I saw that there was a commit last year that added support for us.

**Reply by ArnieO** [reply, marked as answer]
It should work fine with 1K too, but you do not need it to be as small as that - around 3K is sufficient. (You do not want to waste power).

This is the P1 interface:
![136756312-1575f6ed-3e92-4841-813a-dc6f81be9ce5](https://github.com/UtilitechAS/amsreader-firmware/assets/10295178/699c55fb-6b35-42c2-b758-1fa9ebdf5a6a)
![135724732-d5ca9a0c-4257-40c5-87b1-72bb95a5deff](https://github.com/UtilitechAS/amsreader-firmware/assets/10295178/6441bb67-b394-4049-ae7d-4b50e9b5052f)

In order to get out any data, pin 2 (Request) must be connected to +5V.

I have not seen a technical specification for the Iskraemeco AM550, but if it complies to the [P1 companion standard](https://www.netbeheernederland.nl/sites/default/files/2024-02/dsmr_5.0.2_p1_companion_standard.pdf) you can draw a maximum of 250 mA from 5V. But that is the steady state current draw, the ESP32 pulls more than that while it transmits. Which is why you need to be cautious on how you use the 5V.
